### ISSUE TITLE
"name.segments" test fails when `numDots` and `numSegs` are both 0

### ISSUE THREAD

**Issue by aryairani** [unisonweb/unison#1620]
@runarorama In `parser-typechecker/tests/Unison/Core/Test/Name.hs`,

```haskell
test :: Test ()
test = scope "name" $ tests
  [ scope "suffixes" $ tests
    [ scope "empty" $ expectEqual (suffixes "") []
    , scope "one namespace" $ expectEqual (suffixes "bar") ["bar"]
    , scope "two namespaces"
      $ expectEqual (suffixes "foo.bar") ["foo.bar", "bar"]
    , scope "multiple namespaces"
      $ expectEqual (suffixes "foo.bar.baz") ["foo.bar.baz", "bar.baz", "baz"]
    , scope "terms named `.`" $ expectEqual (suffixes "base..") ["base..", "."]
    ]
  , scope "segments" $ do
    numDots <- int' 0 10
    numSegs <- int' 0 10
    n       <- int' 0 10
    segs <- listOf n . pick $ replicate numDots "." ++ replicate numSegs "foo"
    expectEqual (segments $ Name . pack $ intercalate "." segs)
                (NameSegment . pack <$> segs)
  ]
```

`pick` crashes when `numDots` and `numSegs` are both 0.  I'm 🧟‍♂️ and I don't currently understand what the `pick` line is trying to generate.  Is it constructing a list of `segs` from a distribution that is `numDots` parts `"."` and `numSegs` parts `"foo"`?

**Comment by runarorama** [comment]
It's trying to generate a name with a ratio numDots:numSegs of `.` segments to `foo` segments.

I'd just modify as follows:

```
    numDots <- int' 1 10
    numSegs <- int' 1 10
```

**Comment by ceedubs** [comment]
This was resolved in #1838 (though not quite in the way that @runarorama suggested).
